src/main/java/com/puppycrawl/tools/checkstyle/checks/header/RegexpHeaderCheck.java

```diff
@@ -77,8 +78,9 @@ public void setMultiLines(int... list) {
     }
 
     @Override
-    protected void processFiltered(File file, List<String> lines) {
+    protected void processFiltered(File file, FileText fileText) {
         final int headerSize = getHeaderLines().size();
+        final List<String> lines = fileText.getLines();
```

Comment:
please avoid usage of temporal object - `List<String> lines` , that is easy refactoring.